framework/src/base/MooseServer.C

```diff
@@ -0,0 +1,529 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MooseServer.h"
+#include "Moose.h"
+#include "InputParameters.h"
+#include "AppFactory.h"
+#include "pcrecpp.h"
+#include <cstdio>
+
+bool
+MooseServer::setup(MooseApp * moose_app)
+{
+  bool pass = true;
+
+  // cache a pointer to the application that set up this server instance
+
+  this->moose_app = moose_app;
+
+  this->is_setup = true;
+
+  return pass;
```

Comment:
why not `return true;`, what is the purpose of the `pass` variable?

Reply:
Good feedback - I just added a commit addressing this.